src/qilib/configuration_helper/instrument_adapter.py

```diff
@@ -100,3 +100,11 @@ def accept(self, visitor: Visitor) -> None:
 
         """
         visitor.visit(self)
+
+    def __str__(self):
+        """Returns string representation for the InstrumentAdapter.
+
+        Returns:
+            String representation for the InstrumentAdapter.
+        """
+        return self._name
```

Comment:
```suggestion
        return '<{}: {}>'.format(type(self).__name__, self.name)
```
This will give some context to the name (e.g. it shows the object is an `InstrumentAdapter` with a specific name). Currently the classname is already embedded in the name (the name is constructred from the class+address) so it seems a bit double, but in the future the name might be arbitrary.